src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java

```diff
@@ -129,20 +129,46 @@ public GhprbPullRequest(GHPullRequest pr,
             logger.log(Level.INFO,
                        "Author of #{0} {1} on {2} not in whitelist!",
                        new Object[] { id, author.getLogin(), reponame });
-            repo.addComment(id, GhprbTrigger.getDscp().getRequestForTestingPhrase());
+            if (!containsComment(pr, getRequestForTestingPhrase())) {
+                repo.addComment(id, getRequestForTestingPhrase());
+            }
         }
-
         logger.log(Level.INFO,
             "Created Pull Request #{0} on {1} by {2} ({3}) updated at: {4} SHA: {5}",
             new Object[] { id, reponame, author.getLogin(), getAuthorEmail(), updated, this.head });
     }
 
+    /**
+     * Checks whether the specific PR contains a comment with the expected body.
+     *
+     * @return true if the PR contains comment with the specified body, otherwise false
+     */
+    private boolean containsComment(GHPullRequest ghPullRequest, String expectedBody) {
+        List<GHIssueComment> prComments = null;
+        try {
+            prComments = ghPullRequest.getComments();
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "Failed to get comments for PR " + ghPullRequest, e);
+            return false;
+        }
+        for (GHIssueComment comment : prComments) {
+            if (comment.getBody() != null && comment.getBody().contains(expectedBody)) {
```

Comment:
i think contains is not the thing we are looking for. the body should exactly match the expected content, no?

Reply:
Good point, I will fix that.